src/core/FactoryMaker.js

```diff
@@ -85,23 +80,21 @@ let FactoryMaker = (function () {
 
     function getSingletonFactory(classConstructor) {
         return function (context) {
-
+            let instance;
             if (context === undefined) {
                 context = {};
             }
-
-            let instance = getSingletonInstance(context, classConstructor.__dashjs_factory_name);
-
             return {
                 getInstance: function () {
-
                     if (instance) {
                         return instance;
+                    } else {
```

Comment:
It might be nice to clean this up to have a single `return instance;` call at the end. You could just switch this to a `if (!instance)` check. That said, thanks for fixing this up!